include/z64subs.h

```diff
@@ -10,16 +10,28 @@ extern Vec3f gOneVec3f;
 
 #define SUBS_TIME_PATHING_ORDER 3
 
+// NOTE: all on screen checks are bugged, and always evaluate to true, even if off screen
+typedef enum {
+    /* 0 */ SUBS_OFFER_MODE_NONE,
+    /* 1 */ SUBS_OFFER_MODE_GET_ITEM,
+    // Furthur modes all deal with offering exhange items
+    /* 2 */ SUBS_OFFER_MODE_NEARBY,
+    /* 3 */ SUBS_OFFER_MODE_ONSCREEN,
+    /* 4 */ SUBS_OFFER_MODE_AUTO,
+    /* 5 */ SUBS_OFFER_MODE_AUTO_TARGETED, // Also checks nearby and on screen
+    /* 6 */ SUBS_OFFER_MODE_AUTO_NEARBY_ONSCREEN,
+    /* 7 */ SUBS_OFFER_MODE_MAX
```

Comment:
This MAX value is a bit weird, because it is used for and'ing bits, it is more all the bits that can be used by this system instead of just the next value.
For example if we added an extra value to this enum (having the value 7), then MAX should have the value 7 too and not 8, because those are the bits that we want to mask.

So, what about renaming this to `SUBS_OFFER_MODE_MASK` (or similar) and defining it as all the values of this enum or'd?

Reply:
Hmm, we've already kind of done this with the race flags (EventInfDogRaceState and WeekEventRegHorseRaceState) so still kind of like using `MAX` as the name. Though I think it would help if I do hard code it to 7, that way it is more noticeable.

Reply:
Agreed, the enum should not be used for masking, it's both not semantically correct, and liable to cause errors. The other question is whether it makes sense to use the MAX as a function argument, and I agree with Anghelo that it doesn't seem to make sense for that either, an ALL define makes more sense for both.

Reply:
Ok, I'll take it out of the enum and make a `SUBS_OFFER_MODE_MASK` define (I'll do the same for the race flags)